src/bootstrap/config.rs

```diff
@@ -466,6 +468,7 @@ impl Config {
             set(&mut config.quiet_tests, rust.quiet_tests);
             set(&mut config.test_miri, rust.test_miri);
             set(&mut config.wasm_syscall, rust.wasm_syscall);
+            config.rust_thinlto = rust.thinlto.unwrap_or(true);
```

Comment:
I think this section will only get hit if `rust` is `Some` (which it may not be), so could the default-true initialization happen sooner?